client/src/featureform/online.py

```diff
@@ -0,0 +1,174 @@
+from abc import ABC, abstractmethod
+from typing import List, Any, Union
+
+import numpy as np
+from .enums import ScalarType
+from .resources import PineconeConfig
+import pinecone
+import uuid
+
+
+class ValueType(ABC):
+    @abstractmethod
+    def scalar(self):
+        pass
+
+    @abstractmethod
+    def is_vector(self):
+        pass
+
+
+class VectorType(ValueType):
+    def __init__(self, scalar_type: ScalarType, dimension: int, is_embedding: bool):
+        self.scalar_type = scalar_type
+        self.dimension = dimension
+        self.is_embedding = is_embedding
+
+    def scalar(self):
+        return self.scalar_type
+
+    def is_vector(self):
```

Comment:
I think the abstraction could be leveraged instead. Static method on ValueType. isVectorType(value).